src/preload.py

```diff
@@ -633,30 +633,43 @@ def find_non_empty_folder_in_path(path, child_dir=""):
 
 
 def find_docker_aufs_root(mountpoint):
-    # We're looking for a /docker/aufs/diff/<xxxxxxxxxxxxx>/ folder with some
+    # We're looking for a /<docker|balena>/aufs/diff/<xxxxxxxxxxxxx>/ folder with some
     # files not starting with a '.'
-    path = os.path.join(mountpoint, "docker", "aufs", "diff")
-    return find_non_empty_folder_in_path(path)
+    for name in ("docker", "balena"):
+        path = os.path.join(mountpoint, name)
+        if os.path.isdir(path):
+            path = os.path.join(path, "aufs", "diff")
```

Comment:
Only a minor tweak, but I believe 
```python
path = os.path.join(mountpoint, name)
if os.path.isdir(path):
     path = os.path.join(path, "aufs", "diff")
```
 could be collapsed to just:
```python
path = os.path.join(mountpoint, name, "aufs", "diff")
if os.path.isdir(path):
```

Hmmm, although of course that _would_ behave slightly different to the code you're using if we have both `/docker/foo` and `/balena/aufs/diff/foo` directories at the same time ;-)  I don't know enough about resinos to know if that'd ever be the case.

(and the same point also applies to the two separate `os.path.join` calls used in `find_docker_overlay2_root`)